edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/TwinManager.cs

```diff
@@ -265,21 +265,26 @@ static void ValidateTwinCollectionSize(TwinCollection collection)
 
         static void ValidateTwinProperties(JToken properties, int currentDepth)
         {
-            foreach (JProperty kvp in ((JObject)properties).Properties())
+            foreach (JProperty kvp in ((JObject) properties).Properties())
             {
                 ValidatePropertyNameAndLength(kvp.Name);
 
                 ValidateValueType(kvp.Name, kvp.Value);
 
-                string s = kvp.Value.ToString();
-                ValidatePropertyValueLength(kvp.Name, s);
-
-                if ((kvp.Value is JValue) && (kvp.Value.Type is JTokenType.Integer))
+                if (kvp.Value is JValue)
```

Comment:
What happens here if `kvp.Value` is null? Does it evaluate to false here?

Reply:
I added a test case with null value; when validating, (kvp.Value is JValue) return true; and kvp.Value.ToString() will return empty string.

Reply:
I guess I'm worried about a null ref exception here. If this evaluates to true will the `if (kvp.Value.Type is JTokenType.Integer)` throw an NRE?

Reply:
this is JTokenType for your reference.

  public enum JTokenType
  {
    /// <summary>No token type has been set.</summary>
    None,
    /// <summary>A JSON object.</summary>
    Object,
    /// <summary>A JSON array.</summary>
    Array,
    /// <summary>A JSON constructor.</summary>
    Constructor,
    /// <summary>A JSON object property.</summary>
    Property,
    /// <summary>A comment.</summary>
    Comment,
    /// <summary>An integer value.</summary>
    Integer,
    /// <summary>A float value.</summary>
    Float,
    /// <summary>A string value.</summary>
    String,
    /// <summary>A boolean value.</summary>
    Boolean,
    /// <summary>A null value.</summary>
    Null,
    /// <summary>An undefined value.</summary>
    Undefined,
    /// <summary>A date value.</summary>
    Date,
    /// <summary>A raw JSON value.</summary>
    Raw,
    /// <summary>A collection of bytes value.</summary>
    Bytes,
    /// <summary>A Guid value.</summary>
    Guid,
    /// <summary>A Uri value.</summary>
    Uri,
    /// <summary>A TimeSpan value.</summary>
    TimeSpan,
  }

Reply:
It will return "JTokenType.Null"; therefore no NRE will be thrown.